OpenDreamRuntime/Procs/Native/DreamProcNativeIcon.cs

```diff
@@ -94,5 +97,22 @@ static class DreamProcNativeIcon {
             iconObj.Height = height;
             return DreamValue.Null;
         }
+
+        [DreamProc("Turn")]
+        [DreamProcParameter("angle", Type = DreamValueType.Float)]
+        public static DreamValue NativeProc_Turn(DreamObject src, DreamObject usr, DreamProcArguments arguments) {
+            DreamValue angleArg = arguments.GetArgument(0, "angle");
+            if (!angleArg.TryGetValueAsFloat(out float angle)) {
+                return new DreamValue(src); // Defaults to input on invalid angle
+            }
+            return _NativeProc_TurnInternal(src, usr, angle);
+        }
+
+        /// <summary> Turns a given icon a given amount of degrees clockwise. </summary>
+        /// <returns> Returns a new icon which has been rotated </returns>
+        public static DreamValue _NativeProc_TurnInternal(DreamObject src, DreamObject usr, float angle) {
```

Comment:
This is the best way I could think of resolving how there needs to be a `DreamProcArguments` passed from the DM Code-Interpreting-Runtime side, but we can just pass floats around in internal C# code. Without having to construct useless argument objects that get immediately dropped.